From 177256ecadfd1eeefdc198aeb0e794b9d0810f19 Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <skardach@marvell.com>
Date: Wed, 24 Jul 2019 16:54:49 +0200
Subject: [PATCH 295/767] octeontx2-af: report RCLK/SCLK values through mbox

commit 486e19481142492b9deea736bd99bcacad2b747c from
git@git.assembla.com:cavium/WindRiver.linux.git

Report RCLK and SCLK frequencies via MBOX_MSG_READY to users.
Values are read from firmware data.
The mbox message values are scaled in MHz because any multiplier
change will never go below 50MHz resolution.

Change-Id: I423e10cd16fd8b7b5c872edde97f88a938f64abe
Signed-off-by: Stanislaw Kardach <skardach@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/13259
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
(cherry picked from commit c5927833062ee1f54acae0c26a4de370df64e494)
Reviewed-on: https://sj1git1.cavium.com/13534
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h | 3 ++-
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c  | 2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index 289c25d821c6..8e42424ec310 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -325,7 +325,8 @@ enum rvu_af_status {
 
 struct ready_msg_rsp {
 	struct mbox_msghdr hdr;
-	u16    sclk_feq;	/* SCLK frequency */
+	u16    sclk_freq;	/* SCLK frequency (in MHz) */
+	u16    rclk_freq;	/* RCLK frequency (in MHz) */
 };
 
 /* Structure for requesting resource provisioning.
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index caf57b6cdd41..987f66972adc 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -1003,6 +1003,8 @@ int rvu_aq_alloc(struct rvu *rvu, struct admin_queue **ad_queue,
 int rvu_mbox_handler_ready(struct rvu *rvu, struct msg_req *req,
 			   struct ready_msg_rsp *rsp)
 {
+	rsp->rclk_freq = rvu->fwdata->rclk;
+	rsp->sclk_freq = rvu->fwdata->sclk;
 	return 0;
 }
 
-- 
2.31.1

